docs/PublicAPI_design.jl

```diff
@@ -0,0 +1,53 @@
+### Public API - supported in all SDKs
+
+# All funcitons should follow DistributedFactorGraphs.jl and we use Julia here to define the signitures.
+
+# Other lanuages should follow DFG.jl/NvaSDK.jl as closely as possible with the following exceptions
+# - labels are Symbols in julia and strings elsewhere.
+# - mutation in julia is done with ! at the end of the function name and not in other languages.
+# - notably C and Rust do not support function overloading and may also be different or more explicit and perhaps using adjectives.
+
+## Structs
+
+## Functions
+
+### Singular `get` 
+
+getVariable(dfg::NavAbilityDFG, label::Symbol)
+
+getFactor(dfg::NavAbilityDFG, label::Symbol)
+
+getBlobentry(dfg::NavAbilityDFG, variableLabel::Symbol, label::Symbol)
+
+getVariableBlobentry(dfg::NavAbilityDFG, variableLabel::Symbol, label::Symbol)
+
+getFactorBlobentry(dfg::NavAbilityDFG, factorLabel::Symbol, label::Symbol)
+
+getGraphBlobentry(dfg::NavAbilityDFG, label::Symbol)
+
+getAgentBlobentry(dfg::NavAbilityDFG, label::Symbol)
+
+getVariableState(dfg::NavAbilityDFG, variableLabel::Symbol, label::Symbol)
```

Comment:
Also note, we currently have `label::Symbol = :default` in julia. Rust favours explicitness in function arguments to align with Rust's philosophy of clarity and avoiding hidden behaviour. We might want to consider the same on a case-by-case basis.